src/substrait/builders/type.py

```diff
@@ -0,0 +1,76 @@
+
+def date(nullable=True):
+    return stt.Type(date=stt.Type.Date(nullability=stt.Type.NULLABILITY_NULLABLE if nullable else stt.Type.NULLABILITY_REQUIRED))
+
+def interval_year(nullable=True):
+    return stt.Type(interval_year=stt.Type.IntervalYear(nullability=stt.Type.NULLABILITY_NULLABLE if nullable else stt.Type.NULLABILITY_REQUIRED))
+
+def interval_day(precision: int, nullable=True):
+    return stt.Type(interval_day=stt.Type.IntervalDay(precision=precision, nullability=stt.Type.NULLABILITY_NULLABLE if nullable else stt.Type.NULLABILITY_REQUIRED))
+
+def interval_compound(precision: int, nullable=True):
+    return stt.Type(interval_compound=stt.Type.IntervalCompound(precision=precision, nullability=stt.Type.NULLABILITY_NULLABLE if nullable else stt.Type.NULLABILITY_REQUIRED))
+
+def uuid(nullable=True):
+    return stt.Type(uuid=stt.Type.UUID(nullability=stt.Type.NULLABILITY_NULLABLE if nullable else stt.Type.NULLABILITY_REQUIRED))
+
+def fixed_char(length: int, nullable=True):
+    return stt.Type(fixed_char=stt.Type.FixedChar(length=length, nullability=stt.Type.NULLABILITY_NULLABLE if nullable else stt.Type.NULLABILITY_REQUIRED))
+
+def var_char(length: int, nullable=True):
+    return stt.Type(var_char=stt.Type.VarChar(length=length, nullability=stt.Type.NULLABILITY_NULLABLE if nullable else stt.Type.NULLABILITY_REQUIRED))
+
+def fixed_binary(length: int, nullable=True):
+    return stt.Type(fixed_binary=stt.Type.FixedBinary(length=length, nullability=stt.Type.NULLABILITY_NULLABLE if nullable else stt.Type.NULLABILITY_REQUIRED))
+
+def decimal(scale: int, precision: int, nullable=True):
+    return stt.Type(decimal=stt.Type.Decimal(scale=scale, precision=precision, nullability=stt.Type.NULLABILITY_NULLABLE if nullable else stt.Type.NULLABILITY_REQUIRED))
+
+# PrecisionTime
+
+def precision_timestamp(precision: int, nullable=True):
+    return stt.Type(precision_timestamp=stt.Type.PrecisionTimestamp(precision=precision, nullability=stt.Type.NULLABILITY_NULLABLE if nullable else stt.Type.NULLABILITY_REQUIRED))
+
+def precision_timestamp_tz(precision: int, nullable=True):
+    return stt.Type(precision_timestamp_tz=stt.Type.PrecisionTimestampTZ(precision=precision, nullability=stt.Type.NULLABILITY_NULLABLE if nullable else stt.Type.NULLABILITY_REQUIRED))
+
+def struct(types: Iterable[stt.Type], nullable=True):
+    return stt.Type(struct=stt.Type.Struct(types=types, nullability=stt.Type.NULLABILITY_NULLABLE if nullable else stt.Type.NULLABILITY_REQUIRED))
+
+def list(type: stt.Type, nullable=True):
+    return stt.Type(list=stt.Type.List(type=type, nullability=stt.Type.NULLABILITY_NULLABLE if nullable else stt.Type.NULLABILITY_REQUIRED))
+
+def map(key: stt.Type, value: stt.Type, nullable=True):
+    return stt.Type(map=stt.Type.Map(key=key, value=value, nullability=stt.Type.NULLABILITY_NULLABLE if nullable else stt.Type.NULLABILITY_REQUIRED))
+
+def named_struct(names: Iterable[str], struct: stt.Type):
```

Comment:
This doesn't return an stt.Type so it probably doesn't belong in this file.  If you want to keep it here it needs a comment so that users don't mistakenly think that it is the same as others in this file.

Reply:
I thought I could get away with it since `NamedStruct` is also defined in type_pb2. Do you think a return type annotation should be enough to show it's different?